ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/KeccakHasher.java

```diff
@@ -38,14 +38,13 @@ public PoWSolution hash(
       final long number,
       final EpochCalculator epochCalc,
       final byte[] prePowHash) {
-    Bytes input = Bytes.wrap(Bytes.wrap(prePowHash), Bytes.ofUnsignedLong(nonce));
 
     MessageDigest digest = KECCAK_256.get();
-    digest.update(input.toArrayUnsafe());
-    byte[] result = digest.digest();
-    Bytes32 resultBytes = Bytes32.wrap(result);
+    digest.update(prePowHash);
```

Comment:
Also just to clarify: prePowHash == keccak256(rlp(unsealed header)) ?

If that's the case I would call this variable `headerHash` since it's much clearer to me what this part is.

Reply:
Correct. Sure, Besu also has a header hash later on, hence why they make sure to call it "prepow". But at this level of the function, it can be named header hash.

Reply:
In that case, let's leave it how it is, unless we can come up with something better not already taken. `intermediateHash` could be a possible name.

Just as long as it's well documented that prewPowHash is that intermediate pre-pow hash, then we're good.